gobblin-config-management/src/main/java/gobblin/config/client/ConfigClient.java

```
@@ -0,0 +1,257 @@
+   * if policy is RAISE_ERROR and this client try to connect to any {@gobblin.config.configstore.ConfigStore} which is not
+   * {@gobblin.config.configstore.ConfigStoreWithStableVersion}, the RuntimeException will be thrown
+   * 
+   * if policy is CACHE_CONFIG_IN_MEMORY and this client try to connect to any {@gobblin.config.configstore.ConfigStore} which is not
+   * {@gobblin.config.configstore.ConfigStoreWithStableVersion}, the consumer need to cache the retrieved configuration in memory,
+   * otherwise, the same query for the same URI against same configuration store may result different result.
+   * @return - the Configuration Client object
+   */
+  public static ConfigClient createConfigClientWithStableVersion(VERSION_STABILITY_POLICY policy) {
+    return new ConfigClient(policy);
+  }
+
+  /**
+   * 
+   * @return the {@ConfigClient} with policy set to VERSION_STABILITY_POLICY.RAISE_ERROR
+   */
+  public static ConfigClient createDefaultConfigClient() {
+    // create with stable versions
+    return createConfigClientWithStableVersion(VERSION_STABILITY_POLICY.RAISE_ERROR);
+  }
+
+  private ConfigStoreAccessor getConfigStoreAccessor(URI uri) throws Exception {
+    if (this.configStoreMap.containsKey(uri)) {
+      return this.configStoreMap.get(uri);
+    }
+
+    ConfigStoreFactory<ConfigStore> csFactory = this.getConfigStoreFactory(uri);
+    ConfigStore cs = csFactory.createConfigStore(uri);
+    
+    if(!(cs instanceof ConfigStoreWithStableVersion)){
+      if(this.policy == VERSION_STABILITY_POLICY.RAISE_ERROR){
+        throw new Exception(String.format("Try to connect to unstable config store ", cs.getStoreURI()));
+      }
+    }
+    
+    // ConfigStoreFactory scheme name could be different than configStore's StoreURI's scheme name
+    URI csRoot = cs.getStoreURI();
+    URI key = new URI(csFactory.getScheme(), csRoot.getAuthority(), csRoot.getPath(), csRoot.getQuery(), csRoot.getFragment());
```

We could just throw an error if the scheme is different. Is there a point to allowing this use case?